File: Binary-Search.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int l=0, r = nums.Length - 1, i, mid;
        while (l<= r){
            // i = (r+l)/2; // if r and l is int.MaxValue this will make bug (overflow) and res will be negative
            i = l+(r-l)/2;
            mid = nums[i];

            if( target < mid)
                r = i-1;
            else if (target >mid)
                l = i+1;
            else
                return i;
        }
        return -1;
        
    }
}

File: LeetCode/Binary-Search.cpp
1class Solution {
2public:
3    int search(vector<int>& nums, int target) {
4        int l =0;
5        int r = nums.size()-1;
6
7        while(l<=r){
8            int mid = l+(r-l)/2;
9            if(target < nums[mid]) r = mid-1;
10            else if(target > nums[mid]) l = mid+1;
11            else return mid;
12            cout<<mid<<endl;
13        }
14        return -1;
15    }
16};